gradle-jdks/src/main/java/com/palantir/gradle/jdks/OrigGradleWrapperCreator.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+public final class OrigGradleWrapperCreator {
+
+    private static final String OLD_CLASS_NAME = "org.gradle.wrapper.GradleWrapperMain";
+    private static final String NEW_CLASS_NAME = "org.gradle.wrapper.OrigGradleWrapper";
```

Comment:
Now we're using asm we can probably get away with calling this `OriginalGradleWrapperMain` rather than having to keep it the same string length.